bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/DreamScreenHandler.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dreamscreen.internal;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.eclipse.smarthome.core.library.types.OnOffType.*;
+import static org.openhab.binding.dreamscreen.internal.DreamScreenBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DreamScreenHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bruce Brouwer - Initial contribution
+ */
+@NonNullByDefault
+public class DreamScreenHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(DreamScreenHandler.class);
+    private final DreamScreenDatagramServer server;
+    private final @Nullable DreamScreenDynamicStateDescriptionProvider descriptionProvider;
+
+    private @Nullable DreamScreenConfiguration config;
+    private int group;
+    private boolean powerOn;
+    private DreamScreenEnumMode powerOnMode = DreamScreenEnumMode.VIDEO; // TODO: Can this be persisted somewhere?
+    private @Nullable InetAddress address;
+
+    public DreamScreenHandler(Thing thing, DreamScreenDatagramServer server,
+            @Nullable DreamScreenDynamicStateDescriptionProvider descriptionProvider) {
+        super(thing);
+        this.server = server;
+        this.descriptionProvider = descriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(DreamScreenConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            boolean thingReachable = true; // <background task with long running initialization here>
+        try {
```

Comment:
Indentation. You can run `mvn spotless:apply` on your binding to fix any style issues.

Reply:
done